geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TemporalTiledRasterLayer.scala

```diff
@@ -87,6 +87,36 @@ class TemporalTiledRasterLayer(
   def mask(geometries: Seq[MultiPolygon]): TiledRasterLayer[SpaceTimeKey] =
     TemporalTiledRasterLayer(zoomLevel, Mask(rdd, geometries, Mask.Options.DEFAULT))
 
+  def mask(
+    aggregatedRDD: RDD[(SpatialKey, Seq[Geometry])],
+    producePartitioner: () => Option[Partitioner],
+    options: Rasterizer.Options
+  ): TiledRasterLayer[SpaceTimeKey] = {
+    val mapTrans = rdd.metadata.layout.mapTransform
+
+    val rekeyedRDD: RDD[(SpatialKey, (SpaceTimeKey, MultibandTile))] =
+      rdd.map { case (k, v) => (k.getComponent[SpatialKey], (k, v)) }
+
+    val joinedRDD: RDD[(SpatialKey, ((SpaceTimeKey, MultibandTile), Seq[Geometry]))] =
+      producePartitioner() match {
+        case Some(p) =>
+          rekeyedRDD.join(aggregatedRDD, p)
+        case None =>
+          rekeyedRDD.join(aggregatedRDD)
+      }
+
+    val maskedRDD: RDD[(SpaceTimeKey, MultibandTile)] =
+      joinedRDD.mapPartitions ({ partition =>
+        partition.map { case (k, ((tk, v), geoms)) =>
+          (tk, v.mask(mapTrans(k), geoms, options))
+        }
+        // TODO: Make sure that preservesPartitioning doesn't negativly impact
```

Comment:
What is the potential concern with it ?

Reply:
The keys of the `RDD` are changing, and it's generally not recommended to have `preservesPartitioning` be `true` if that's the case.

Reply:
Ah, right. Thats correct thats concern in general, specifically that when performing `.mapPartitions(???, preservesPartitioning=true)` you could change the value of the key such that it would no longer map to the partition on which it resides. However, because in this case the RDD is result of the join on the keys directly above and the keys are not changed in the `mapPartitions` we're totally safe. Essentially `k == tk` is always true.